hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/SecurityConfig.java

```diff
@@ -123,10 +122,8 @@ public SecurityConfig(ConfigurationSource configuration) {
 
     // Please Note: To make it easy for our customers we will attempt to read
     // HDDS metadata dir and if that is not set, we will use Ozone directory.
-    // TODO: We might want to fix this later.
-    this.metadatDir = this.configuration.get(HDDS_METADATA_DIR_NAME,
-        configuration.get(OZONE_METADATA_DIRS,
-            configuration.get(HDDS_DATANODE_DIR_KEY)));
+    this.metadataDir = this.configuration.get(HDDS_METADATA_DIR_NAME,
+        configuration.get(OZONE_METADATA_DIRS));
```

Comment:
We can add a null check here, to log the error where it originates, for the `metadataDir `should both the `hdds.metadata.dir` ozone-site property and the fallback `ozone.metadata.dirs `is not set, such as
 https://github.com/apache/ozone/blob/fe748d6776fc33124dad2d6cfc5f5dceee038a7f/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/hdds/datanode/metadata/CRLDBDefinition.java#L77

Reply:
There are already null checks for this in `SecurityConfig`. Check [getKeyLocation()](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/SecurityConfig.java#L273-L277) and [getCertificateLocation()](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/SecurityConfig.java#L286-L290). 

With the new change and without `ozone.metadata.dirs` specified, that's the output in the logs
```
2022-11-11 14:45:16,642 [main] INFO ha.HASecurityUtils: Initializing secure StorageContainerManager.
Metadata directory can't be null. Please check configs.
2022-11-11 14:45:16,992 [shutdown-hook-0] INFO server.StorageContainerManagerStarter: SHUTDOWN_MSG: 
/************************************************************
SHUTDOWN_MSG: Shutting down StorageContainerManager at scm/172.22.0.10

```

Reply:
We could make the log more specific so that it includes the class.